src/foam/demos/examples/services.fbe

```diff
@@ -0,0 +1,436 @@
+        return task;
+      `
+    },
+    {
+      name: 'deleteTask',
+      async: true, 
+      type: 'boolean',
+      args: 'String taskId',
+      javaCode: `
+        boolean existed = getTaskMap().containsKey(taskId);
+        getTaskMap().remove(taskId);
+        System.out.println("Task deleted: " + taskId + " " + (existed ? "success" : "not found"));
+        return existed;
+      `
+    },
+    {
+      name: 'getTaskMap',
+      type: 'java.util.Map',
+      javaCode: `
+        // In real implementation, use DAO or database
+        if ( taskMap_ == null ) {
+          taskMap_ = new java.util.concurrent.ConcurrentHashMap<>();
+        }
+        return taskMap_;
+      `
+    }
+  ],
+  
+  properties: [
+    {
+      class: 'Object',
+      name: 'taskMap',
+      javaType: 'java.util.Map<String, Task>',
+      visibility: 'HIDDEN'
+    }
+  ]
+});
+
+// Define Task model
+foam.CLASS({
+  package: 'foam.demo.service',
+  name: 'Task',
+  
+  properties: [
+    { class: 'String', name: 'id' },
+    { class: 'String', name: 'title' },
+    { class: 'String', name: 'description' },
+    { class: 'DateTime', name: 'created' }
+  ]
+});
+
+log('TaskService interface with default Java implementation defined');
+
+##  Separate Java Implementation
+
+For more complex services, create a separate Java file implementing the interface:
+--
+// ServerTaskService.java
+/*
+package foam.demo.service;
+
+import foam.core.ContextAwareSupport;
+import foam.dao.DAO;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.Map;
+
+public class ServerTaskService extends ContextAwareSupport implements TaskService {
+  
+  private final Map<String, Task> taskMap = new ConcurrentHashMap<>();
+  
+  @Override
+  public String createTask(String title, String description) {
+    String taskId = "task-" + System.currentTimeMillis();
+    Task task = (Task) Task.create(getX())
+      .setId(taskId)
+      .setTitle(title)
+      .setDescription(description)
+      .setCreated(new java.util.Date());
+    
+    taskMap.put(taskId, task);
+    
+    // In real implementation, use DAO:
+    // DAO taskDAO = (DAO) getX().get("taskDAO");
+    // taskDAO.put(task);
+    
+    getLogger().info("Task created: " + taskId);
+    return taskId;
+  }
+  
+  @Override
+  public Task getTask(String taskId) {
+    Task task = taskMap.get(taskId);
+    getLogger().info("Task retrieved: " + taskId + " " + (task != null ? "found" : "not found"));
+    return task;
+  }
+  
+  @Override
+  public boolean deleteTask(String taskId) {
+    boolean existed = taskMap.containsKey(taskId);
+    taskMap.remove(taskId);
+    getLogger().info("Task deleted: " + taskId + " " + (existed ? "success" : "not found"));
+    return existed;
+  }
+}
+*/
+
+log('Separate Java implementation example shown above');
+
+##  Service Registration
+
+Register services in services.jrl using CSpec (Component Specification):
+--
+// Example services.jrl entry:
+/*
```

Comment:
Think you are still missing CSpec setup for TaskDAO, which the TaskService interacts with.